src/v8/v8.cc

```diff
@@ -24,14 +24,19 @@
 #include <utility>
 #include <vector>
 
+#include "v8.h"
 #include "v8-version.h"
 #include "wasm-api/wasm.hh"
 
 namespace proxy_wasm {
 namespace {
 
 wasm::Engine *engine() {
-  static const auto engine = wasm::Engine::make();
+  static wasm::own<wasm::Engine> engine;
+  if (engine.get() == nullptr) {
+    v8::V8::EnableWebAssemblyTrapHandler(true);
+    engine = wasm::Engine::make();
+  }
```

Comment:
I'm worried that if this is thread-safe.

Reply:
Do you mean Trap Handler or the singleton?

Reply:
I mean accessing this singleton. I guess `engine()` can be called on any worker thread?

Reply:
Yes, but we access it first on the main thread, so on the worker threads it's never going to enter that `if` block.

Reply:
OK, makes sense. But it seems to me that this cpp-host depends on the host-specific (Envoy-specific) implementation detail.. I would prefer use std::call_once or similar to guard this block. WDYT?

Reply:
Done.